src/main/java/Duke.java

```diff
@@ -1,10 +1,67 @@
+import duke.exception.DukeException;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+import java.util.Scanner;
+
+/**
+ * Duke is a chat based task manager that keeps track of the Tasks keyed in by the user.
+ */
 public class Duke {
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+    private Parser parser;
+
+    /**
+     * Constructs a Duke object.
+     */
+    public Duke() {
+        String filePath = "data/duke.txt";
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+            parser = new Parser(tasks);
+        } catch (DukeException e) {
+            ui.showError(e.toString());
+            tasks = new TaskList();
+            parser = new Parser(tasks);
+        }
+    }
+
+    /**
+     * Starts Duke.
+     */
+    public void run() {
+        boolean isExit = false;
+        Scanner sc = new Scanner(System.in);
+
+        System.out.println(ui.showWelcome());
+
+        while(!isExit) {
+            String input = sc.nextLine();
+            if(input.equals("bye")) {
+                System.out.println(parser.parse(input));
```

Comment:
Why not consolidate duplicate code?

Reply:
By that do you mean to abstract out the System.out.println()? Could you suggest how I can go about doing that? Thanks!

Reply:
I am referring to line 47 and 50 can be simplified to one line above the if else clause.

Reply:
Oh ok I see what you meant, thanks!